pallets/author-mapping/src/lib.rs

```diff
@@ -32,29 +32,167 @@ pub use pallet::*;
 pub mod pallet {
 
 	use frame_support::pallet_prelude::*;
+	use frame_support::traits::ReservableCurrency;
 	use frame_system::pallet_prelude::*;
 	use nimbus_primitives::AccountLookup;
+	use sp_runtime::Percent;
+
+	/// The portion of the security deposit that goes to the the account who reports it occupying
+	/// space after it should have been cleaned or rotated.
+	pub const NARC_REWARD: Percent = Percent::from_percent(5);
+
+	/// The period of time after which an AuthorId can be reported as defunct.
+	/// This value should be roughly the recommended key rotation period.
+	pub const NARC_GRACE_PERIOD: u32 = 2_000;
+
+	/// The security deposit amount.
+	pub const DEPOSIT_AMOUNT: u32 = 500;
 
 	#[pallet::pallet]
 	pub struct Pallet<T>(PhantomData<T>);
 
-	/// Configuration trait of this pallet.
+	/// Configuration trait of this pallet. We tightly couple to Parachain Staking in order to
```

Comment:
So I wouldn't be able to register my keys before joining as a candidate. I guess it is safe enough to consider there will be enough time to inject the keys and be selected as a collator. 
(Should we enforce that a candidate has a key registered before selecting as a collator ?)